resolver-dns/src/main/java/io/netty/resolver/dns/DnsQueryContext.java

```diff
@@ -78,15 +85,20 @@ DnsQuestion question() {
         return question;
     }
 
+    List<DnsRecord> additional() {
+        return additional;
+    }
+
     void query() {
         final DnsQuestion question = question();
+        final List<DnsRecord> additional = additional();
         final InetSocketAddress nameServerAddr = nameServerAddr();
         final DatagramDnsQuery query = new DatagramDnsQuery(null, nameServerAddr, id);
         query.setRecursionDesired(recursionDesired);
-        query.setRecord(DnsSection.QUESTION, question);
-        if (optResource != null) {
-            query.setRecord(DnsSection.ADDITIONAL, optResource);
-        }
+
+        addRecord(query, DnsSection.QUESTION, question);
+        addRecords(query, DnsSection.ADDITIONAL, additional);
+        addRecord(query, DnsSection.ADDITIONAL, optResource);
```

Comment:
I'm not sure there's value in `addRecord` if it's only called there. My own choice would be to directly write the if test here, like 
```java
if (optResource != null) {
  query.addRecord(DnsSection.ADDITIONAL, optResource);
}
```

But  let's the committers decide :)